generators/entity-server/templates/src/main/java/package/web/rest/EntityResource.java.ejs

```diff
@@ -86,7 +99,8 @@ import static org.elasticsearch.index.query.QueryBuilders.*;<% } %>
  * REST controller for managing <%= entityClass %>.
  */
 @RestController
-@RequestMapping("/api")
+@RequestMapping("/api")<% if (isUsingMapsId === true && !viaService) { %>
+@Transactional <% } %>
```

Comment:
Not sure we want @Transactional in Resource!

Reply:
Thanks for reviewing.

Two things:
1. This annotation will be added only in case where we use entity directly from Resource Controller.(option 1 during question)
2. Without this annotation, `detached entity passed to persist` will occur in `createXXX` as I have to load the parent entity to get its id.

Reply:
In all cases, this is bad to have @Transactional anywhere in the controller

Reply:
Yes, you are correct. It is bad. But since we are serving the entity directly from controller, I can not see other option. One stupid thing can be done is annotate only the `createXXX` as only this method is needed `@Transactional`
For the entities managed via service class will not required to do this.

I am open to discuss and receive better suggestions.

Reply:
I am not sure how behaves jHipster when no service and no DTO are selected in option -> is the entity directly returned in the controller (domain model)?
If not mistaken, DTOs require Service to be true

Reply:
Yes, entity is returned directly in that case.

Reply:
Thanks @DanielFran! Good point.